src/material/autocomplete/autocomplete.ts

```diff
@@ -127,8 +128,9 @@ export class MatAutocomplete extends _MatAutocompleteMixinBase implements AfterC
   /** @docs-private */
   @ContentChildren(MatOption, {descendants: true}) options: QueryList<MatOption>;
 
+  // TODO: Remove cast once https://github.com/angular/angular/pull/37506 is available.
   /** @docs-private */
-  @ContentChildren(MatOptgroup, {descendants: true}) optionGroups: QueryList<MatOptgroup>;
+  @ContentChildren(MAT_OPTGROUP as any, {descendants: true}) optionGroups: QueryList<MatOptgroup>;
```

Comment:
I'm wondering whether this would have an effect, considering that `MatOptgroup` has a hard reference in `MatOptionModule` which is then referenced in `MatAutocompleteModule`. Since we have the size checks set up, it should be easy to find out.

Reply:
I'll double-check, but I was thinking "yes" (given the size in the test also decreased: https://github.com/angular/components/pull/19577/commits/e63d9ddef1e3308f1f9ed8a12004107aa9d567c2). My thinking is that the module information is just used for the compilation resolution, but the compiler will just reference the directives/components based on usage. Providers are persisted in the module injector definition.

Reply:
Okay, double-checked. Everything works as expected. The module definitions by default will not reference declarations etc. (as noted above). The compiler will only reference these when `emitInline` is set. This is only the case in JIT when it gets through the compiler facade.